plugin/trino-opa/src/test/java/io/trino/plugin/opa/TestOpaBatchAccessControlFiltering.java

```diff
@@ -49,226 +52,235 @@
-                TEST_SECURITY_CONTEXT,
-                "my_catalog",
-                requestedColumns);
-
-        Set<String> expectedRequests = Stream.of("table_one", "table_two", "table_three")
-                .map(tableName ->
-                        """
-                        {
-                            "operation": "FilterColumns",
-                            "filterResources": [
-                                {
-                                    "table": {
-                                        "tableName": "%s",
-                                        "schemaName": "my_schema",
-                                        "catalogName": "my_catalog",
-                                        "columns": ["%s_column_one", "%s_column_two"]
-                                    }
-                                }
-                            ]
-                        }
-                        """.formatted(tableName, tableName, tableName))
+
+        BiFunction<Integer, String, Set<String>> buildColumnSet = (numberOfColumns, tableName) -> IntStream.rangeClosed(1, numberOfColumns)
+                .mapToObj(column -> "%s_column_%s".formatted(tableName, column))
                 .collect(toImmutableSet());
-        assertStringRequestsEqual(expectedRequests, mockClient.getRequests(), "/input/action");
-        assertThat(result).containsExactlyInAnyOrderEntriesOf(
-                ImmutableMap.<SchemaTableName, Set<String>>builder()
-                        .put(tableOne, ImmutableSet.of("table_one_column_one", "table_one_column_two"))
-                        .put(tableTwo, ImmutableSet.of("table_two_column_two"))
-                        .buildOrThrow());
+
+        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.of(
+                tableOne, buildColumnSet.apply(12, "table_one"),
+                tableTwo, buildColumnSet.apply(10, "table_two"),
+                tableThree, buildColumnSet.apply(7, "table_three"));
+
+        List<TestCase> testCases = ImmutableList.of(new TestCase(OPA_CONFIG_NO_BATCH_SIZE, ImmutableMap.of(12, 1, 10, 1, 7, 1), 3),
+                new TestCase(OPA_CONFIG_BATCH_SIZE_2, ImmutableMap.of(2, 6 + 5 + 3, 1, 1), 6 + 5 + 4),
+                new TestCase(OPA_CONFIG_BATCH_SIZE_3, ImmutableMap.of(3, 4 + 3 + 2, 1, 2), 4 + 4 + 3),
+                new TestCase(OPA_CONFIG_BATCH_SIZE_100, ImmutableMap.of(12, 1, 10, 1, 7, 1), 3)
+        );
+
+        for (TestCase testCase : testCases) {
+            InstrumentedHttpClient mockClient = createMockHttpClient(
+                    OPA_SERVER_BATCH_URI,
+                    buildHandler(pathToResources,
+                            ImmutableSet.of("\"table_one_column_1\"", "\"table_one_column_2\"", "\"table_one_column_3\"", "\"table_two_column_1\"")
+                                    .stream()
+                                    .map(RequestTestUtilities::toJsonNode)
+                                    .collect(toImmutableSet())));
```

Comment:
nit: we could add a `toJsonNode` util that takes an object so we don't have to double quote the string